packages/xsnap/src/xsnap.js

```diff
@@ -310,6 +310,7 @@ export function xsnap(options) {
    */
   async function close() {
     baton = baton.then(async () => {
+      await messagesToXsnap.next(encoder.encode(`q`));
```

Comment:
This uses the explicit "quit" message that was introduced in https://github.com/agoric-labs/xsnap-pub/pull/33